docs/datatypes/semaphore.md

```diff
@@ -3,14 +3,14 @@ id: datatypes_semaphore
 title:  "Semaphore"
 ---
 
-A `Semaphore` datatype which allows synchronization between fibers with `acquire` and `release` operations.
+A `Semaphore` datatype which allows synchronization between fibers with the `withPermit` operation, which safely acquires and releases a permit.
```

Comment:
Note that it is still possible to trick the system by asking for a negative number of permits.

Reply:
Wouldn't `assertNonNegative` cause the fiber to die at that point?

Reply:
Yes, on release, which seems wrong 🤔 It's like catching a bad guy when it's too late 😆

Reply:
I see. Yes we could definitely fix that.